### PR TITLE
CI Changes - Move to using tox

### PR DESCRIPTION
**Pull request by gauravojha**

### CODE REVIEW

**Comment by gauravojha** [comment]
Changes being merged:
- move to using `tox` to run unit tests
- added Makefile for ease of usage

**Comment by gauravojha** [comment]
@tumb1er  please review when you get a chance.

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,2 @@
+test:
+	tox
```
Comment (tumb1er) [inline Makefile]: Should here be a newline at the end?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[tox]
+envlist = py3
+
+[testenv]
+deps =
+  coverage==4.5.4
+commands =
+  sh scripts/update_assets.sh
+  sh scripts/update_assets.sh
```
Comment (tumb1er) [inline tox.ini:10]: A comment is needed here that second update_assets call is not a typo.

**Review by tumb1er — CHANGES_REQUESTED** [review]

File: tox.ini (hunk 2 above)
Comment (gauravojha) [reply tox.ini:10]: @tumb1er  thanks. Yup. makes sense, let me add a comment there, but I don't have a detailed context on the reasons behind this.

**Comment by tumb1er** [comment]
@gauravojha Would you like replace Python-3.4 with Python-3.8 here? Or in separate PR?

**Comment by gauravojha** [comment]
@tumb1er  I was thinking separate PR, and keeping the context of this PR only on moving to `tox`, but either is fine with me. What would you prefer?

**Comment by tumb1er** [comment]
> What would you prefer?

Separate PR is more clear.

File: tox.ini (hunk 2 above)
Comment (tumb1er) [reply tox.ini:10]: Just for your information: this scripts runs a test suite that generates coverage file on the same code base that checks this file, so first run may fail after codebase changes, and the second one should be fixed (in theory at least).

**Review by tumb1er — APPROVED** [review]

**Pull request merged**